coral-dbt/src/main/resources/macros/coral_macros/spark/utils/generate_incremental_code.sql

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+{% macro spark__generate_incremental_code(coral_response, table_names, output_table) -%}
+
+--     Compiled lines of spark scala code to be executed, separated by \n delimiter
+  {% set spark_scala = '' %}
+
+--     Ex.
+--       sql: SELECT * FROM db.t1 UNION SELECT * FROM db.t2
+--       table_names: ['db.t1', 'db.t2']
+--
+--       incremental_maintenance_sql: SELECT * FROM db_t1_delta UNION SELECT * FROM db_t2_delta
+--       underscore_delimited_table_names: ['db_t1', 'db_t2']
+--       incremental_table_names: ['db_t1_delta', 'db_t2_delta']
+  {% set incremental_maintenance_sql = coral_response['incremental_maintenance_sql'] %}
+  {% set underscore_delimited_table_names = coral_response['underscore_delimited_table_names'] %}
+  {% set incremental_table_names = coral_response['incremental_table_names'] %}
+
+--     Generate Iceberg incremental table scan code for each target table
+--     Namespace variable used to persist changes beyond loop scope
```

Comment:
also add a comment here to elaborate on how the start and end snapshot ids are retrieved and that logic might change in the future.